lib/Redmine/Api/AbstractApi.php

```diff
@@ -96,7 +96,7 @@ protected function delete($path)
      */
     protected function isNotNull($var)
     {
-        return !is_null($var);
+        return !(is_null($var) || $var === false || $var === "" || ((is_array($var) || is_object($var)) && empty($var)));
```

Comment:
I will update the condition formatting, but unfortunately yours won't work. The last condition (the one for object and arrays) should be like this:
```
!( (is_array($var) || is_object($var)) && empty($var) )
```

I'll commit the changes in a sec.